proof.go

```diff
@@ -113,25 +116,27 @@ func (proof Proof) VerifyNamespace(h hash.Hash, nID namespace.ID, data [][]byte,
 		}
 		return false
 	}
-	gotLeafHashes := make([][]byte, 0, len(data))
+	// TODO [Me] we could also check whether the number of leaves match the proof range end-start and make an early return if not
+	gotLeafHashes := make([][]byte, 0, len(leaves))
 	nIDLen := nID.Size()
 	if proof.IsOfAbsence() {
 		gotLeafHashes = append(gotLeafHashes, proof.leafHash)
 	} else {
-		// collect leaf hashes from provided data and
+		// collect leaf hashes from provided leaves and
 		// do some sanity checks:
 		hashLeafFunc := nth.HashLeaf
-		for _, gotLeaf := range data {
+		for _, gotLeaf := range leaves {
+			// TODO [Me] can be converted to something like isNameSpaceIDPrefixed()
 			if len(gotLeaf) < int(nIDLen) {
 				// conflicting namespace sizes
 				return false
 			}
-			gotLeafNid := namespace.ID(gotLeaf[:nIDLen])
+			gotLeafNid := namespace.ID(gotLeaf[:nIDLen]) // TODO [Me] a helper function
```

Comment:
👍🏼 everything that improves readability (without impacting performance much) should be done.